### PR TITLE
Fix clone method

### PR DESCRIPTION
**Pull request by Torgan**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,11 +17,11 @@
     "type": "library",
     "license": "Apache-2.0",
     "require": {
-        "php": ">=5.6",
+        "php": "^7.1",
         "symfony/options-resolver": "^2.7|^3.0|^4.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^4.8"
+        "phpunit/phpunit": "^7.0"
```
Comment (ooflorent) [inline composer.json:24]: Why not version 8?

File: src/Type/Base/ConstrainedObject.php [hunk 2]
```diff
@@ -310,21 +306,42 @@ public function setAttribute($name, $value)
...
-                is_null($value))
-            ) {
+            if ($classDefinition) {
+
                 $class = $classDefinition->getClass();
 
+                if (is_null($value)) {
+                    if (
+                        ($classDefinition->isNullable() === false) &&
+                        is_null($value)
```
Comment (ooflorent) [inline src/Type/Base/ConstrainedObject.php]: Already tested. You don’t need this.

File: src/Type/Base/BasicObject.php [hunk 3]
```diff
@@ -316,12 +316,13 @@ public function __debugInfo()
      */
     public function __clone()
     {
-        $class = $this->getClass();
-
-        return new $class($this);
+        foreach ($this->attribute as $key => $value) {
+            if (is_object($value)) {
+                $this->$key = clone($value);
```
Comment (ooflorent) [inline src/Type/Base/BasicObject.php]: `clone $value`

File: composer.json (hunk 1 above)
Comment (Torgan) [reply composer.json]: Have to upgrade php on my computer

File: src/Type/Base/ConstrainedObject.php (hunk 2 above)
Comment (Torgan) [reply src/Type/Base/ConstrainedObject.php]: Indeed

File: composer.json (hunk 1 above)
Comment (Torgan) [reply composer.json:24]: Since we still have servers using php7.1 (www), it's safer to still test for this configuration.

Comment (Torgan) [reply composer.json:24]: And phpunit 8 requires php 7.2

**Review by Ciloe — APPROVED** [review]

**Pull request merged**